ResearchKit/SurveyTasks/ORKEpworthSleepScale.m

```diff
@@ -0,0 +1,124 @@
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ The Epworth Sleep Scale is a validated tool to assess somnolence. High Epworth scores are associated
+ With diseases such as obstructive sleep apnoea.
+ 
+ If you would like to read more about this research tool, visit the original publication article:
+ 
+ http://www.mwjohns.com/wp-content/uploads/2009/murray_papers/reliabiltiy_and_factor_analysis_of_the_epworth_sleepiness_scale.pdf
+ */
+
+
+#import "ORKEpworthSleepScale.h"
+#import "ORKDefines_Private.h"
+#import "ORKOrderedTask.h"
+#import "ORKCompletionStep.h"
+
+@implementation ORKEpworthSleepScale
+
+static void ORKStepArrayAddStep(NSMutableArray *array, ORKStep *step) {
+    [step validateParameters];
+    [array addObject:step];
+}
+
+- (instancetype)initWithIdentifier:(NSString *)identifier {
+    
+    NSMutableArray *steps = [NSMutableArray array];
+    
+    // Explain the purpose of the introduction survey
+    {
+        ORKInstructionStep *step = [[ORKInstructionStep alloc] initWithIdentifier:@"IntroStep"];
+        step.title = ORKLocalizedString(@"EPWORTH_INTRO_TITLE", nil);
+        step.text = ORKLocalizedString(@"EPWORTH_INTRO_TEXT", nil);
+        ORKStepArrayAddStep(steps, step);
+    }
+    
+    NSArray *answerChoices = [NSArray arrayWithObjects:
+                              [[ORKTextChoice alloc] initWithText:ORKLocalizedString(@"EPWORTH_ANSWER_0", nil) detailText:nil value:[NSNumber numberWithInt:0] exclusive:true],
+                              [[ORKTextChoice alloc] initWithText:ORKLocalizedString(@"EPWORTH_ANSWER_1", nil) detailText:nil value:[NSNumber numberWithInt:1] exclusive:true],
+                              [[ORKTextChoice alloc] initWithText:ORKLocalizedString(@"EPWORTH_ANSWER_2", nil) detailText:nil value:[NSNumber numberWithInt:2] exclusive:true],
+                              [[ORKTextChoice alloc] initWithText:ORKLocalizedString(@"EPWORTH_ANSWER_3", nil) detailText:nil value:[NSNumber numberWithInt:3] exclusive:true],
+                              nil];
+    
+    ORKTextChoiceAnswerFormat *answerFormat = [[ORKTextChoiceAnswerFormat alloc] initWithStyle:ORKChoiceAnswerStyleSingleChoice textChoices:answerChoices];
+    
+    for (int questionID = 0; questionID < 8; questionID++) {
+        ORKQuestionStep *step = [[ORKQuestionStep alloc] initWithIdentifier:[NSString stringWithFormat:@"%d",questionID]];
+            step.title = ORKLocalizedString(@"EPWORTH_INTRO_TITLE", nil);
+        {
+            NSString *questionStem = [NSString stringWithFormat:@"EPWORTH_Q%d",questionID];
+            step.text = ORKLocalizedString(questionStem, nil);
+            step.optional = NO;
+        }
+        step.answerFormat = answerFormat;
+        ORKStepArrayAddStep(steps,step);
+    }
+    
+    ORKCompletionStep *finalStep = [[ORKCompletionStep alloc] initWithIdentifier:@"FinalStep"];
```

Comment:
Yeah agreed. I will add completion steps to all the surveys but put the results on an instruction step. That makes more sense I think :-).